src/Entities/Assertions/AssertionFactory.php

```diff
@@ -314,18 +315,27 @@ public function getInstance(\stdClass $annotation)
             case Requires::ANNOTATION:
                 // complex annotations leave us with two possibilities: raw or custom assertions
 
-                if (isset($annotation->values['type'])) {
+                if (isset($annotation->values['type'], $annotation->values['constraint'])) {
                     // we need a custom assertion here
 
+                    $potentialAssertion = $annotation->values['type'];
+                    if (class_exists($potentialAssertion)) {
+                        $assertionInstance = new $potentialAssertion($annotation->values['constraint']);
+                        if (!$assertionInstance instanceof AssertionInterface) {
+                            throw new \Exception('Assertions have to include the AssertionInterface');
```

Comment:
COuld you please be more specific? Like sprintf('Specified assertion type %s does not implement %s', potentialAssertion, AssertionInterface::class);